scan.c

```diff
@@ -126,20 +134,49 @@ struct iio_scan_context * iio_create_scan_context(
 		return NULL;
 	}
 
-	if (!backend || strstr(backend, "local"))
-		ctx->scan_local = true;
+	ctx->backendopts = iio_strndup(backend ? backend : "local:usb:ip:", PATH_MAX);
+	if (!ctx->backendopts) {
+		free(ctx);
+		errno = ENOMEM;
+		return NULL;
+	}
 
-	if (!backend || strstr(backend, "usb"))
-		ctx->scan_usb = true;
+	/* while we hope people use usb=vid,pid ; iit's common to use
+	 * usb=vid:pid ; so fix it up
+	 */
+	ptr = ctx->backendopts;
+	while ((ptr = strstr(ptr, "usb="))) {
+		uint32_t val;
+
+		val = strtoul(&ptr[sizeof("usb=") - 1], &ptr2, 16);
+		if (val < 0xFFFF && val > 0 && ptr2) {
+			if (ptr2[0] == ':' ) {
+				if (ptr2[1] == '*' || ptr2[1] == '\0') {
+					ptr2[0] = ',';
+				} else {
+					val = strtoul(&ptr2[1], &ptr3, 16);
+					if (val < 0xFFFF && val > 0) {
+						ptr2[0] = ',';
+						ptr = ptr3 - sizeof("usb=");
+					}
+				}
+			}
 
-	if (!backend || strstr(backend, "ip"))
-		ctx->scan_network = true;
+		}
+		ptr = ptr + sizeof("usb=");
+	}
 
 	return ctx;
 }
 
 void iio_scan_context_destroy(struct iio_scan_context *ctx)
 {
+	if (!ctx)
+		return;
+
+	if (ctx->backendopts)
+		free(ctx->backendopts);
+
 	free(ctx);
```

Comment:
`free()` handles NULL pointers, so you don't need to check for ctx and ctx->backendopts. Besides... Is there a single case where this function will be called with ctx->backendopts being NULL?